src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,13 @@
+import Popup from "./Popup.js";
+
+export class PopupWithConfirmation extends Popup{
+  constructor(popupSelector, handleConfirm){
+    super(popupSelector);
+    this._confimButton = this._popupElement.querySelector('.modal__button')
+    this._handleConfirm = handleConfirm;
+  }
+  setEventListeners() {
+    super.setEventListeners();
+    this._handleConfirm;
+  }
+}
```

Comment:
* When I delete 3 cards in a row, there are errors https://snipboard.io/10gDW9.jpg 
*  Now all your cards (which you've tried to delete but changed your mind) are **deleted** all at once. Please try it yourself: add 3 cards in a row then try to delete them, but don't do it (cancel and close the confirmation popup) and then delete any other card - and all your previous cards are deleted at once (all 3 and the other one). It’s because of a **memory leak**. All listeners are gathered in one element and they are fired all together when you confirm deleting. You need either to **correctly remove** submit listeners or update the card data. 

You should implemente `setSubmitHandler` here to reassign it on opening